prow/pubsub/subscriber/subscriber.go

```diff
@@ -161,25 +180,32 @@ func (s *Subscriber) handlePeriodicJob(l *logrus.Entry, msg messageInterface, su
 	if periodicJob == nil {
 		err := fmt.Errorf("failed to find associated periodic job %s", pe.Name)
 		l.WithError(err).Errorf("failed to create job %s", pe.Name)
+		prowJob = pjutil.NewProwJobWithAnnotation(kube.ProwJobSpec{}, nil, pe.Annotations)
+		reportProwJobFailure(err)
 		return err
 	}
 	prowJobSpec := pjutil.PeriodicSpec(*periodicJob)
-	var prowJob kube.ProwJob
-	// Add annotations
+	// Adds / Updates Labels from prow job event
+	for k, v := range pe.Labels {
+		periodicJob.Labels[k] = v
+	}
+
+	// Adds annotations
 	prowJob = pjutil.NewProwJobWithAnnotation(prowJobSpec, periodicJob.Labels, pe.Annotations)
```

Comment:
Interesting, slightly surprised that prowJob in reportProwJobFailure gets the right value (but it does!):

https://play.golang.org/p/QCgX93gHkTG

I'd be happier if the signature was `reportProwJobFailure(pj *ProwJob, err error)`

Reply:
done